scripts/pre-commit

```diff
@@ -0,0 +1 @@
+make fmtcheck
```

Comment:
I'd make the precommit hook simply run `gofmt` to automatically format the code. After all, if the code isn't formatted, this script will exit with an error, and the *only* thing the developer can do is run `gofmt` anyway, so why not do it for them?